### PR TITLE
Session dialog improvement

### PR DESCRIPTION
**Pull request by fat23cat**

### CODE REVIEW

File: ddp-workspace/projects/toolkit/src/lib/components/dialogs/sessionWillExpire.component.ts
Comment (fat23cat) [inline ddp-workspace/projects/toolkit/src/lib/components/dialogs/sessionWillExpire.component.ts]: If a user clicked 'Continue' or 'Sign Out' buttons Auth0 needs some time to handle the user's request to refresh the session or redirect the user to the home page. While Auth0 is handling the request, time may end even and will be sent one more request. To prevent it, let's check whether the user chose something or not.

**Review by MocanaAtBroad — COMMENTED** [review]
Maybe better to hold off on this until we merge the session bug fix PR?

**Comment by fat23cat** [comment]
> Maybe better to hold off on this until we merge the session bug fix PR?

Definitely agree, for this reason, I converted it to draft.

File: ddp-workspace/projects/ddp-sdk/src/lib/services/renewSessionNotifier.service.ts
Comment (fat23cat) [inline ddp-workspace/projects/ddp-sdk/src/lib/services/renewSessionNotifier.service.ts:11]: Found a bug:
1. Session dialog was shown 
2. A user reloads a page 
3. In half the cases, dialog won't be shown after page reloading. 
I suppose the reason for the bug was that we subscribed to `openDialogEvents` after the event happened, so we heed to save the last event.

File: ddp-workspace/projects/toolkit/src/lib/components/dialogs/sessionWillExpire.component.ts
Comment (fat23cat) [inline ddp-workspace/projects/toolkit/src/lib/components/dialogs/sessionWillExpire.component.ts:29]: With a slow Internet connection, the token renewal process can take some time, let's show to a user that something happens.

**Review by MocanaAtBroad — APPROVED** [review]
Works really nicely! And the progress bar is a very nice touch.

**Pull request merged**
